### PR TITLE
Increase the timeout for the resilience tests

### LINKED ISSUE #2632: Resilience tests are flaky in the 002 environment
## Issue Description
After enabling address sanitizer in the 002 environment, the `resilience tests` test case is failing a majority of the time. This test has always been flaky, but address sanitizer might be causing it to fail more.

I've also been able to reproduce this locally sometimes by running `src/stellar-core test 'resilience tests'` without address sanitizer.

## Expected Result
Test should pass.

## Actual Result
Resilience tests will fail with an error like - 

```
simulation/CoreTests.cpp:157: FAILED:
  {Unknown expression after the reported line}
due to unexpected exception with messages:
  restart victim 0
  Simulation timed out
```

### PR DESCRIPTION
**Pull request by hidenori-shinohara**
# Description

Resolves #2632
tl; dr: Things seem to be behaving the way they should be, but it's taking a little longer than one might initially expect. Increasing the timeout will fix the issue.

* The only test in the resilience tests that seems to fail after trying 200 random seeds (0, 1, ..., 199) is `custom-A`.
* `custom-A` is a network with 6 nodes. It removes one node from the network, puts that node back in the network. It again removes and puts that node in the network. In other words, there are `6 * 6 = 36` possibilities.
* After iterating through all the possibilities, I found out that there seemed to be only two possible combinations that fail the test: `(5, 3)` and `(2, 3)`. They can be reproduced by
  * `src/stellar-core test '[resilience]' -c 'custom-A' --rng-seed 82`
  * `src/stellar-core test '[resilience]' -c 'custom-A' --rng-seed 174`
* By examining the log obtained through `--ll DEBUG`, the only things that take more than 5 seconds are:
  * Nominate delta such as `2020-07-30T09:22:11.108 <test> [Herder DEBUG] Nominate delta for slot 9 is 28100 ms`
  * Externalize delay such as `2020-07-30T09:22:11.242 <test> [Herder DEBUG] externalize delay (GB3X5) delta for slot 9 is 12400 ms`

1. The `Nominate delta` seems to be related to #2106.
2. All the long externalize delays happen only when a "victim" node tries to close a ledger that was closed while it was away. For instance, the `12400 ms` above is calculated by `(when GB3X5 closed slot 9) - (the earliest time slot 9 was externalized)` and the number is so big because `(the earliest time slot 9 was externalized)` happened while `GB3x5` was away from the network. Checking `steady_clock` shows that there is a very small delay between a victim re-joining and externalizing.

Therefore, I decided to increase the timeout. The multiplier of 4 still fails the custom-A test, but with 5, the resilience tests pass with 10k seeds (1, 2, ..., 10,000), which implies that the test probably has the passing rate > 99.99%.

# Checklist
- [x] Reviewed the [contributing](https://github.com/stellar/stellar-core/blob/master/CONTRIBUTING.md#submitting-changes) document
- [x] Rebased on top of master (no merge commits)
- [x] Ran `clang-format` v5.0.0 (via `make format` or the Visual Studio extension)
- [x] Compiles
- [x] Ran all tests
- [x] If change impacts performance, include supporting evidence per the [performance document](https://github.com/stellar/stellar-core/blob/master/performance-eval/performance-eval.md)

### CODE REVIEW

**Comment by marta-lokhova** [comment]
As we're touching up on customA topology, do you mind adding a few clarification comments to `Topologies::customA`? Specifically, we should explain why threshold=4 is okay for A,B,C (we do not expect byzantine faults), and why node `I` is never included in the simulation (`I` simulates a dead node)

**Comment by MonsieurNicolas** [comment]
I am merging this now, we keep running into this timeout which stops us from getting a clean read on a build

**Comment by MonsieurNicolas** [comment]
@hidenori-shinohara you can address Marta's comment in a separate PR

**Comment by MonsieurNicolas** [comment]
r+ 3199ff24ca80d3a95f1ea064745dcf88d82e2126
